### PR TITLE
Refactor code to make `score_metric` flexible

### LINKED ISSUE #107: Use phenodigm_score to populate the top level score, add score_metric
Rather than promoting ancestor_information_content as the top level score for a best match, it would make more sense to promote the phenodigm_score. In addition, it makes sense to to add something like `"score_metric":"phenodigm"` alongside the score to make it explicit, and allow for an option to change which score is promoted up as an overall score.   

```
  "subject_best_matches": {
    "MP:0002169": {
      "match_source": "MP:0002169",
      "match_source_label": "no abnormal phenotype detected (MPO)",
      "match_target": "HP:0004325",
      "match_target_label": "Decreased body weight (HPO)",
      "score": 1.4431977534690428,
      "match_subsumer": null,
      "match_subsumer_label": null,
      "similarity": {
        "subject_id": "MP:0002169",
        "subject_label": null,
        "subject_source": null,
        "object_id": "HP:0004325",
        "object_label": null,
        "object_source": null,
        "ancestor_id": "UPHENO:0001003",
        "ancestor_label": "",
        "ancestor_source": null,
        "object_information_content": null,
        "subject_information_content": null,
        "ancestor_information_content": 1.4431977534690428,
        "jaccard_similarity": 0.16216216216216217,
        "cosine_similarity": null,
        "dice_similarity": null,
        "phenodigm_score": 0.48376861011243283
      }
    },
```

Should look more like
```
  "subject_best_matches": {
    "MP:0002169": {
      "match_source": "MP:0002169",
      "match_source_label": "no abnormal phenotype detected (MPO)",
      "match_target": "HP:0004325",
      "match_target_label": "Decreased body weight (HPO)",
      "score": 0.48376861011243283,
      "score_metric": "phenodigm"
      "match_subsumer": null,
      "match_subsumer_label": null,
      "similarity": {
        "subject_id": "MP:0002169",
        "subject_label": null,
        "subject_source": null,
        "object_id": "HP:0004325",
        "object_label": null,
        "object_source": null,
        "ancestor_id": "UPHENO:0001003",
        "ancestor_label": "",
        "ancestor_source": null,
        "object_information_content": null,
        "subject_information_content": null,
        "ancestor_information_content": 1.4431977534690428,
        "jaccard_similarity": 0.16216216216216217,
        "cosine_similarity": null,
        "dice_similarity": null,
        "phenodigm_score": 0.48376861011243283
      }
    },

```

### PR DESCRIPTION
**Pull request by hrshdhgd**
This PR adds `score_metric`/`metric` as a param to determine the ranking of searches.

fixes #107

### CODE REVIEW

**Comment by hrshdhgd** [comment]
The new tsps looks like this if metric chosen is `phenodigm_score`:

```
tsps = TermsetPairwiseSimilarity {
    subject_termset: [
        {
            "MP:0005389": {
                "id": "MP:0005389",
                "label": "reproductive system phenotype (MPO)",
            },
        },
        {
            "MP:0005391": {
                "id": "MP:0005391",
                "label": "vision/eye phenotype (MPO)",
            },
        },
        {
            "MP:0010771": {
                "id": "MP:0010771",
                "label": "integument phenotype (MPO)",
            },
        },
        {
            "MP:0005367": {
                "id": "MP:0005367",
                "label": "renal/urinary system phenotype (MPO)",
            },
        },
        {
            "MP:0002169": {
                "id": "MP:0002169",
                "label": "no abnormal phenotype detected (MPO)",
            },
        },
    ],
    subject_best_matches: {
        "MP:0002169": {
            "match_source": "MP:0002169",
            "match_source_label": "no abnormal phenotype detected (MPO)",
            "match_target": "MP:0006144",
            "match_target_label": "increased systemic arterial systolic blood pressure (MPO)",
            "score": "0.7437442427684151",
            "score_metric": "phenodigm_score",
        },
        "MP:0005367": {
            "match_source": "MP:0005367",
            "match_source_label": "renal/urinary system phenotype (MPO)",
            "match_target": "HP:0000093",
            "match_target_label": "Proteinuria (HPO)",
            "score": "1.123043316723227",
            "score_metric": "phenodigm_score",
        },
        "MP:0005389": {
            "match_source": "MP:0005389",
            "match_source_label": "reproductive system phenotype (MPO)",
            "match_target": "MP:0006144",
            "match_target_label": "increased systemic arterial systolic blood pressure (MPO)",
            "score": "1.124435603086443",
            "score_metric": "phenodigm_score",
        },
        "MP:0005391": {
            "match_source": "MP:0005391",
            "match_source_label": "vision/eye phenotype (MPO)",
            "match_target": "MP:0006144",
            "match_target_label": "increased systemic arterial systolic blood pressure (MPO)",
            "score": "1.094445249016253",
            "score_metric": "phenodigm_score",
        },
        "MP:0010771": {
            "match_source": "MP:0010771",
            "match_source_label": "integument phenotype (MPO)",
            "match_target": "MP:0006144",
            "match_target_label": "increased systemic arterial systolic blood pressure (MPO)",
            "score": "1.1590416901826253",
            "score_metric": "phenodigm_score",
        },
    },
    subject_best_matches_similarity_map: {
        "MP:0002169": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.12962962962962962",
            "object_id": "MP:0006144",
            "phenodigm_score": "0.7437442427684151",
            "subject_id": "MP:0002169",
        },
        "MP:0005367": {
            "ancestor_id": "UBERON:0001008",
            "ancestor_information_content": "6.936744601801884",
            "ancestor_label": "renal system",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.18181818181818182",
            "object_id": "HP:0000093",
            "phenodigm_score": "1.123043316723227",
            "subject_id": "MP:0005367",
        },
        "MP:0005389": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.2962962962962963",
            "object_id": "MP:0006144",
            "phenodigm_score": "1.124435603086443",
            "subject_id": "MP:0005389",
        },
        "MP:0005391": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.2807017543859649",
            "object_id": "MP:0006144",
            "phenodigm_score": "1.094445249016253",
            "subject_id": "MP:0005391",
        },
        "MP:0010771": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.3148148148148148",
            "object_id": "MP:0006144",
            "phenodigm_score": "1.1590416901826253",
            "subject_id": "MP:0010771",
        },
    },
    object_termset: [
        {
            "MP:0006144": {
                "id": "MP:0006144",
                "label": "increased systemic arterial systolic blood pressure (MPO)",
            },
        },
        {
            "HP:0004325": {
                "id": "HP:0004325",
                "label": "Decreased body weight (HPO)",
            },
        },
        {
            "HP:0000093": {
                "id": "HP:0000093",
                "label": "Proteinuria (HPO)",
            },
        },
    ],
    object_best_matches: {
        "HP:0000093": {
            "match_source": "HP:0000093",
            "match_source_label": "Proteinuria (HPO)",
            "match_target": "MP:0005367",
            "match_target_label": "renal/urinary system phenotype (MPO)",
            "score": "1.123043316723227",
            "score_metric": "phenodigm_score",
        },
        "HP:0004325": {
            "match_source": "HP:0004325",
            "match_source_label": "Decreased body weight (HPO)",
            "match_target": "MP:0005367",
            "match_target_label": "renal/urinary system phenotype (MPO)",
            "score": "0.8704058181327751",
            "score_metric": "phenodigm_score",
        },
        "MP:0006144": {
            "match_source": "MP:0006144",
            "match_source_label": "increased systemic arterial systolic blood pressure (MPO)",
            "match_target": "MP:0010771",
            "match_target_label": "integument phenotype (MPO)",
            "score": "1.1590416901826253",
            "score_metric": "phenodigm_score",
        },
    },
    object_best_matches_similarity_map: {
        "HP:0000093": {
            "ancestor_id": "UBERON:0001008",
            "ancestor_information_content": "6.936744601801884",
            "ancestor_label": "renal system",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.18181818181818182",
            "object_id": "MP:0005367",
            "phenodigm_score": "1.123043316723227",
            "subject_id": "HP:0000093",
        },
        "HP:0004325": {
            "ancestor_id": "UBERON:0000468",
            "ancestor_information_content": "2.2728188647181566",
            "ancestor_label": "multicellular organism",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.3333333333333333",
            "object_id": "MP:0005367",
            "phenodigm_score": "0.8704058181327751",
            "subject_id": "HP:0004325",
        },
        "MP:0006144": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.3148148148148148",
            "object_id": "MP:0010771",
            "phenodigm_score": "1.1590416901826253",
            "subject_id": "MP:0006144",
        },
    },
    average_score: 4.646681455846709,
    best_score: 1.1590416901826253,
    metric: PhenodigmScore,
}
Time taken for second termset_pairwise_similarity: 9.006227ms
[src/lib.rs:2276:9] &_tsps = TermsetPairwiseSimilarity {
    subject_termset: [
        {
            "MP:0002169": {
                "id": "MP:0002169",
                "label": "no abnormal phenotype detected (MPO)",
            },
        },
        {
            "MP:0005367": {
                "id": "MP:0005367",
                "label": "renal/urinary system phenotype (MPO)",
            },
        },
        {
            "MP:0005389": {
                "id": "MP:0005389",
                "label": "reproductive system phenotype (MPO)",
            },
        },
        {
            "MP:0010771": {
                "id": "MP:0010771",
                "label": "integument phenotype (MPO)",
            },
        },
        {
            "MP:0005391": {
                "id": "MP:0005391",
                "label": "vision/eye phenotype (MPO)",
            },
        },
    ],
    subject_best_matches: {
        "MP:0002169": {
            "match_source": "MP:0002169",
            "match_source_label": "no abnormal phenotype detected (MPO)",
            "match_target": "MP:0006144",
            "match_target_label": "increased systemic arterial systolic blood pressure (MPO)",
            "score": "0.7437442427684151",
            "score_metric": "phenodigm_score",
        },
        "MP:0005367": {
            "match_source": "MP:0005367",
            "match_source_label": "renal/urinary system phenotype (MPO)",
            "match_target": "HP:0000093",
            "match_target_label": "Proteinuria (HPO)",
            "score": "1.123043316723227",
            "score_metric": "phenodigm_score",
        },
        "MP:0005389": {
            "match_source": "MP:0005389",
            "match_source_label": "reproductive system phenotype (MPO)",
            "match_target": "MP:0006144",
            "match_target_label": "increased systemic arterial systolic blood pressure (MPO)",
            "score": "1.124435603086443",
            "score_metric": "phenodigm_score",
        },
        "MP:0005391": {
            "match_source": "MP:0005391",
            "match_source_label": "vision/eye phenotype (MPO)",
            "match_target": "MP:0006144",
            "match_target_label": "increased systemic arterial systolic blood pressure (MPO)",
            "score": "1.094445249016253",
            "score_metric": "phenodigm_score",
        },
        "MP:0010771": {
            "match_source": "MP:0010771",
            "match_source_label": "integument phenotype (MPO)",
            "match_target": "MP:0006144",
            "match_target_label": "increased systemic arterial systolic blood pressure (MPO)",
            "score": "1.1590416901826253",
            "score_metric": "phenodigm_score",
        },
    },
    subject_best_matches_similarity_map: {
        "MP:0002169": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.12962962962962962",
            "object_id": "MP:0006144",
            "phenodigm_score": "0.7437442427684151",
            "subject_id": "MP:0002169",
        },
        "MP:0005367": {
            "ancestor_id": "UBERON:0001008",
            "ancestor_information_content": "6.936744601801884",
            "ancestor_label": "renal system",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.18181818181818182",
            "object_id": "HP:0000093",
            "phenodigm_score": "1.123043316723227",
            "subject_id": "MP:0005367",
        },
        "MP:0005389": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.2962962962962963",
            "object_id": "MP:0006144",
            "phenodigm_score": "1.124435603086443",
            "subject_id": "MP:0005389",
        },
        "MP:0005391": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.2807017543859649",
            "object_id": "MP:0006144",
            "phenodigm_score": "1.094445249016253",
            "subject_id": "MP:0005391",
        },
        "MP:0010771": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.3148148148148148",
            "object_id": "MP:0006144",
            "phenodigm_score": "1.1590416901826253",
            "subject_id": "MP:0010771",
        },
    },
    object_termset: [
        {
            "HP:0000093": {
                "id": "HP:0000093",
                "label": "Proteinuria (HPO)",
            },
        },
        {
            "HP:0004325": {
                "id": "HP:0004325",
                "label": "Decreased body weight (HPO)",
            },
        },
        {
            "MP:0006144": {
                "id": "MP:0006144",
                "label": "increased systemic arterial systolic blood pressure (MPO)",
            },
        },
    ],
    object_best_matches: {
        "HP:0000093": {
            "match_source": "HP:0000093",
            "match_source_label": "Proteinuria (HPO)",
            "match_target": "MP:0005367",
            "match_target_label": "renal/urinary system phenotype (MPO)",
            "score": "1.123043316723227",
            "score_metric": "phenodigm_score",
        },
        "HP:0004325": {
            "match_source": "HP:0004325",
            "match_source_label": "Decreased body weight (HPO)",
            "match_target": "MP:0005367",
            "match_target_label": "renal/urinary system phenotype (MPO)",
            "score": "0.8704058181327751",
            "score_metric": "phenodigm_score",
        },
        "MP:0006144": {
            "match_source": "MP:0006144",
            "match_source_label": "increased systemic arterial systolic blood pressure (MPO)",
            "match_target": "MP:0010771",
            "match_target_label": "integument phenotype (MPO)",
            "score": "1.1590416901826253",
            "score_metric": "phenodigm_score",
        },
    },
    object_best_matches_similarity_map: {
        "HP:0000093": {
            "ancestor_id": "UBERON:0001008",
            "ancestor_information_content": "6.936744601801884",
            "ancestor_label": "renal system",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.18181818181818182",
            "object_id": "MP:0005367",
            "phenodigm_score": "1.123043316723227",
            "subject_id": "HP:0000093",
        },
        "HP:0004325": {
            "ancestor_id": "UBERON:0000468",
            "ancestor_information_content": "2.2728188647181566",
            "ancestor_label": "multicellular organism",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.3333333333333333",
            "object_id": "MP:0005367",
            "phenodigm_score": "0.8704058181327751",
            "subject_id": "HP:0004325",
        },
        "MP:0006144": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.3148148148148148",
            "object_id": "MP:0010771",
            "phenodigm_score": "1.1590416901826253",
            "subject_id": "MP:0006144",
        },
    },
    average_score: 4.646681455846709,
    best_score: 1.1590416901826253,
    metric: PhenodigmScore,
}
```

Now, `average_score` is still average AIC score. Is this acceptable?

**Comment by justaddcoffee** [comment]
Great, thanks @hrshdhgd! 

I think we want the average_score to use whatever metric the user specified in the metric arg, don’t we?

**Comment by caufieldjh** [comment]
I would also expect `average_score` to correspond to the selected metric - there isn't a way for users to know that it's actually average AIC otherwise.

Have you seen any performance hits with the TSPS using phenodigm?

**Comment by hrshdhgd** [comment]
> I would also expect `average_score` to correspond to the selected metric - there isn't a way for users to know that it's actually average AIC otherwise.

I added code to show the corresponding average score now.

> Have you seen any performance hits with the TSPS using phenodigm?
1 extra ms in benchmarks ... so not so much.

**Comment by hrshdhgd** [comment]
```
result = TermsetPairwiseSimilarity {
    subject_termset: [
        {
            "MP:0005391": {
                "id": "MP:0005391",
                "label": "vision/eye phenotype (MPO)",
            },
        },
        {
            "MP:0005389": {
                "id": "MP:0005389",
                "label": "reproductive system phenotype (MPO)",
            },
        },
        {
            "MP:0005367": {
                "id": "MP:0005367",
                "label": "renal/urinary system phenotype (MPO)",
            },
        },
        {
            "MP:0002169": {
                "id": "MP:0002169",
                "label": "no abnormal phenotype detected (MPO)",
            },
        },
        {
            "MP:0010771": {
                "id": "MP:0010771",
                "label": "integument phenotype (MPO)",
            },
        },
    ],
    subject_best_matches: {
        "MP:0002169": {
            "match_source": "MP:0002169",
            "match_source_label": "no abnormal phenotype detected (MPO)",
            "match_target": "MP:0006144",
            "match_target_label": "increased systemic arterial systolic blood pressure (MPO)",
            "score": "0.7437442427684151",
            "score_metric": "phenodigm_score",
        },
        "MP:0005367": {
            "match_source": "MP:0005367",
            "match_source_label": "renal/urinary system phenotype (MPO)",
            "match_target": "HP:0000093",
            "match_target_label": "Proteinuria (HPO)",
            "score": "1.123043316723227",
            "score_metric": "phenodigm_score",
        },
        "MP:0005389": {
            "match_source": "MP:0005389",
            "match_source_label": "reproductive system phenotype (MPO)",
            "match_target": "MP:0006144",
            "match_target_label": "increased systemic arterial systolic blood pressure (MPO)",
            "score": "1.124435603086443",
            "score_metric": "phenodigm_score",
        },
        "MP:0005391": {
            "match_source": "MP:0005391",
            "match_source_label": "vision/eye phenotype (MPO)",
            "match_target": "MP:0006144",
            "match_target_label": "increased systemic arterial systolic blood pressure (MPO)",
            "score": "1.094445249016253",
            "score_metric": "phenodigm_score",
        },
        "MP:0010771": {
            "match_source": "MP:0010771",
            "match_source_label": "integument phenotype (MPO)",
            "match_target": "MP:0006144",
            "match_target_label": "increased systemic arterial systolic blood pressure (MPO)",
            "score": "1.1590416901826253",
            "score_metric": "phenodigm_score",
        },
    },
    subject_best_matches_similarity_map: {
        "MP:0002169": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.12962962962962962",
            "object_id": "MP:0006144",
            "phenodigm_score": "0.7437442427684151",
            "subject_id": "MP:0002169",
        },
        "MP:0005367": {
            "ancestor_id": "UBERON:0001008",
            "ancestor_information_content": "6.936744601801884",
            "ancestor_label": "renal system",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.18181818181818182",
            "object_id": "HP:0000093",
            "phenodigm_score": "1.123043316723227",
            "subject_id": "MP:0005367",
        },
        "MP:0005389": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.2962962962962963",
            "object_id": "MP:0006144",
            "phenodigm_score": "1.124435603086443",
            "subject_id": "MP:0005389",
        },
        "MP:0005391": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.2807017543859649",
            "object_id": "MP:0006144",
            "phenodigm_score": "1.094445249016253",
            "subject_id": "MP:0005391",
        },
        "MP:0010771": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.3148148148148148",
            "object_id": "MP:0006144",
            "phenodigm_score": "1.1590416901826253",
            "subject_id": "MP:0010771",
        },
    },
    object_termset: [
        {
            "HP:0004325": {
                "id": "HP:0004325",
                "label": "Decreased body weight (HPO)",
            },
        },
        {
            "MP:0006144": {
                "id": "MP:0006144",
                "label": "increased systemic arterial systolic blood pressure (MPO)",
            },
        },
        {
            "HP:0000093": {
                "id": "HP:0000093",
                "label": "Proteinuria (HPO)",
            },
        },
    ],
    object_best_matches: {
        "HP:0000093": {
            "match_source": "HP:0000093",
            "match_source_label": "Proteinuria (HPO)",
            "match_target": "MP:0005367",
            "match_target_label": "renal/urinary system phenotype (MPO)",
            "score": "1.123043316723227",
            "score_metric": "phenodigm_score",
        },
        "HP:0004325": {
            "match_source": "HP:0004325",
            "match_source_label": "Decreased body weight (HPO)",
            "match_target": "MP:0005367",
            "match_target_label": "renal/urinary system phenotype (MPO)",
            "score": "0.8704058181327751",
            "score_metric": "phenodigm_score",
        },
        "MP:0006144": {
            "match_source": "MP:0006144",
            "match_source_label": "increased systemic arterial systolic blood pressure (MPO)",
            "match_target": "MP:0005391",
            "match_target_label": "vision/eye phenotype (MPO)",
            "score": "1.094445249016253",
            "score_metric": "phenodigm_score",
        },
    },
    object_best_matches_similarity_map: {
        "HP:0000093": {
            "ancestor_id": "UBERON:0001008",
            "ancestor_information_content": "6.936744601801884",
            "ancestor_label": "renal system",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.18181818181818182",
            "object_id": "MP:0005367",
            "phenodigm_score": "1.123043316723227",
            "subject_id": "HP:0000093",
        },
        "HP:0004325": {
            "ancestor_id": "UBERON:0000468",
            "ancestor_information_content": "2.2728188647181566",
            "ancestor_label": "multicellular organism",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.3333333333333333",
            "object_id": "MP:0005367",
            "phenodigm_score": "0.8704058181327751",
            "subject_id": "HP:0004325",
        },
        "MP:0006144": {
            "ancestor_id": "MP:0000001",
            "ancestor_information_content": "4.26719956102326",
            "ancestor_label": "mammalian phenotype (MPO)",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.2807017543859649",
            "object_id": "MP:0005391",
            "phenodigm_score": "1.094445249016253",
            "subject_id": "MP:0006144",
        },
    },
    average_score: 1.0498861476841341,
    best_score: 1.1590416901826253,
    metric: PhenodigmScore,
}
```

**Comment by hrshdhgd** [comment]
```
tsps = TermsetPairwiseSimilarity {
    subject_termset: [
        {
            "GO:0016020": {
                "id": "GO:0016020",
                "label": "membrane",
            },
        },
        {
            "GO:0005634": {
                "id": "GO:0005634",
                "label": "nucleus",
            },
        },
    ],
    subject_best_matches: {
        "GO:0005634": {
            "match_source": "GO:0005634",
            "match_source_label": "nucleus",
            "match_target": "GO:0031965",
            "match_target_label": "nuclear membrane",
            "score": "0.6956521739130435",
            "score_metric": "jaccard_similarity",
        },
        "GO:0016020": {
            "match_source": "GO:0016020",
            "match_source_label": "membrane",
            "match_target": "GO:0031965",
            "match_target_label": "nuclear membrane",
            "score": "0.34782608695652173",
            "score_metric": "jaccard_similarity",
        },
    },
    subject_best_matches_similarity_map: {
        "GO:0005634": {
            "ancestor_id": "GO:0005634",
            "ancestor_information_content": "5.8496657269155685",
            "ancestor_label": "nucleus",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.6956521739130435",
            "object_id": "GO:0031965",
            "phenodigm_score": "2.017258704180859",
            "subject_id": "GO:0005634",
        },
        "GO:0016020": {
            "ancestor_id": "GO:0016020",
            "ancestor_information_content": "4.8496657269155685",
            "ancestor_label": "membrane",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.34782608695652173",
            "object_id": "GO:0031965",
            "phenodigm_score": "1.298784144051735",
            "subject_id": "GO:0016020",
        },
    },
    object_termset: [
        {
            "GO:0005773": {
                "id": "GO:0005773",
                "label": "vacuole",
            },
        },
        {
            "GO:0031965": {
                "id": "GO:0031965",
                "label": "nuclear membrane",
            },
        },
    ],
    object_best_matches: {
        "GO:0005773": {
            "match_source": "GO:0005773",
            "match_source_label": "vacuole",
            "match_target": "GO:0005634",
            "match_target_label": "nucleus",
            "score": "0.8333333333333334",
            "score_metric": "jaccard_similarity",
        },
        "GO:0031965": {
            "match_source": "GO:0031965",
            "match_source_label": "nuclear membrane",
            "match_target": "GO:0005634",
            "match_target_label": "nucleus",
            "score": "0.6956521739130435",
            "score_metric": "jaccard_similarity",
        },
    },
    object_best_matches_similarity_map: {
        "GO:0005773": {
            "ancestor_id": "GO:0043231",
            "ancestor_information_content": "5.112700132749362",
            "ancestor_label": "intracellular membrane-bounded organelle",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.8333333333333334",
            "object_id": "GO:0005634",
            "phenodigm_score": "2.06411807897654",
            "subject_id": "GO:0005773",
        },
        "GO:0031965": {
            "ancestor_id": "GO:0005634",
            "ancestor_information_content": "5.8496657269155685",
            "ancestor_label": "nucleus",
            "cosine_similarity": "NaN",
            "jaccard_similarity": "0.6956521739130435",
            "object_id": "GO:0005634",
            "phenodigm_score": "2.017258704180859",
            "subject_id": "GO:0031965",
        },
    },
    average_score: 0.6878019323671498,
    best_score: 0.8333333333333334,
    metric: JaccardSimilarity,
}
```

File: src/similarity.rs
[the remark this answers is not in the document]
Comment (justaddcoffee) [reply src/similarity.rs:1]: thanks! it's a bit of clutter, but future us might thank us for this

**Pull request merged**